pom.xml

```diff
@@ -99,7 +99,7 @@
     <antlr-input.dir>src/main/antlr3</antlr-input.dir>
     <antlr-output.dir>target/generated-sources/antlr3</antlr-output.dir>
     <test.output.tofile>true</test.output.tofile>
-    <top.dir>${project.basedir}</top.dir>
+    <top.dir>${project.basedir}/..</top.dir>
```

Comment:
top.dir in the submodules doesn't seem to do anything.
I suggest that you delete it from those to simplify the maven setup.